File: Basic-Video-Chat-ConnectionService_FCM/app/src/main/java/com/tokbox/sample/basicvideochat_connectionservice/NotificationChannelManager.java
package com.tokbox.sample.basicvideochat_connectionservice;

import static androidx.core.content.ContextCompat.getSystemService;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.media.AudioAttributes;
import android.media.RingtoneManager;
import android.net.Uri;
import android.os.Build;

import androidx.annotation.RequiresApi;

public class NotificationChannelManager {

    public static final String INCOMING_CALL_CHANNEL_ID = "vonage_video_call_channel";
    public static final String ONGOING_CALL_CHANNEL_ID = "vonage_ongoing_video_call_channel";

    private Context context;

    public NotificationChannelManager(Context context) {
        this.context = context;
    }


    @RequiresApi(api = Build.VERSION_CODES.O)
    void setup() {
        setupIncomingCallChannel();
        setupOngoingCallChannel();
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private void setupIncomingCallChannel() {
        NotificationChannel channel = new NotificationChannel(INCOMING_CALL_CHANNEL_ID, "Incoming Calls",
                NotificationManager.IMPORTANCE_HIGH);

        Uri ringtoneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_RINGTONE);
        channel.setSound(ringtoneUri, new AudioAttributes.Builder()
                .setUsage(AudioAttributes.USAGE_NOTIFICATION_RINGTONE)
                .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
                .build());

        NotificationManager mgr = context.getSystemService(NotificationManager.class);
        mgr.createNotificationChannel(channel);
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private void setupOngoingCallChannel() {
        NotificationChannel channel = new NotificationChannel(ONGOING_CALL_CHANNEL_ID, "Incoming Calls",
                NotificationManager.IMPORTANCE_DEFAULT);

        NotificationManager mgr = context.getSystemService(NotificationManager.class);
        mgr.createNotificationChannel(channel);
    }
}

File: Basic-Video-Chat-ConnectionService-Kotlin/app/src/main/java/com/tokbox/sample/basicvideochatconnectionservice/NotificationChannelManager.kt
package com.tokbox.sample.basicvideochatconnectionservice

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.media.AudioAttributes
import android.media.RingtoneManager
import android.os.Build
import androidx.annotation.RequiresApi

class NotificationChannelManager(private val context: Context) {

    @RequiresApi(api = Build.VERSION_CODES.O)
    fun setup() {
        setupIncomingCallChannel()
        setupOngoingCallChannel()
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private fun setupIncomingCallChannel() {
        val channel = NotificationChannel(
            INCOMING_CALL_CHANNEL_ID, "Incoming Calls",
            NotificationManager.IMPORTANCE_HIGH
        )

        val ringtoneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_RINGTONE)
        channel.setSound(
            ringtoneUri, AudioAttributes.Builder()
                .setUsage(AudioAttributes.USAGE_NOTIFICATION_RINGTONE)
                .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
                .build()
        )

        val mgr = context.getSystemService(
            NotificationManager::class.java
        )
        mgr.createNotificationChannel(channel)
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private fun setupOngoingCallChannel() {
        val channel = NotificationChannel(
            ONGOING_CALL_CHANNEL_ID, "Ongoing Calls",
            NotificationManager.IMPORTANCE_DEFAULT
        )

        val mgr = context.getSystemService(
            NotificationManager::class.java
        )
        mgr.createNotificationChannel(channel)
    }

    companion object {
        const val INCOMING_CALL_CHANNEL_ID: String = "vonage_video_call_channel"
        const val ONGOING_CALL_CHANNEL_ID: String = "vonage_ongoing_video_call_channel"
    }
}
